presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -527,20 +527,26 @@ public PlanOptimizers(
                 new CheckSubqueryNodesAreRewritten(),
                 new StatsRecordingPlanOptimizer(
                         optimizerStats,
-                        new PredicatePushDown(metadata, typeOperators, typeAnalyzer, false, false)),
-                new IterativeOptimizer(
-                        ruleStats,
-                        statsCalculator,
-                        estimatedExchangesCostCalculator,
-                        ImmutableSet.<Rule<?>>builder()
-                                .addAll(columnPruningRules)
-                                .addAll(projectionPushdownRules)
-                                .add(new RemoveRedundantIdentityProjections())
-                                .add(new PushLimitIntoTableScan(metadata))
-                                .add(new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))
-                                .add(new PushSampleIntoTableScan(metadata))
-                                .add(new PushAggregationIntoTableScan(metadata))
-                                .build()),
+                        new PredicatePushDown(metadata, typeOperators, typeAnalyzer, false, false)));
+
+        IterativeOptimizer pushIntoTableScanOptimizer = new IterativeOptimizer(
```

Comment:
i thought about that. "at the top" is a few screens away, which makes code navigation harder
it would be definitely better if `pushIntoTableScanOptimizer` was used in more than one place (i consider current two incantations still  "one place")